### PR TITLE
LibJS: Symbol cleanup

### PR DESCRIPTION
**Pull request by linusg**

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/VM.h [hunk 1]
```diff
@@ -264,7 +264,8 @@ class VM : public RefCounted<VM> {
 
     StackInfo m_stack_info;
 
-    HashMap<DeprecatedString, Symbol*> m_global_symbol_map;
+    // https://tc39.es/ecma262/#table-globalsymbolregistry-record-fields
```
Comment (davidot) [inline Userland/Libraries/LibJS/Runtime/VM.h:271]: Is it intentional to only have the link and not the actual name in the comment?

**Review by davidot — COMMENTED** [review]
Just one comment, also I feel like the first two commits can be merged but :shrug: it's fine

File: Userland/Libraries/LibJS/Runtime/VM.h (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/VM.h]: grey area as `GlobalSymbolMap` doesn't have its own section link in the spec, but name + fields table works. Thanks!

**Comment by linusg** [comment]
> also I feel like the first two commits can be merged

If it was really just touching the declaration I probably would've agreed, but since I had to change some of the pointer types as well it felt like too many unrelated changes for a rename commit :)

**Pull request merged**